### PR TITLE
Add github action for test and build

### PR DESCRIPTION
**Pull request by Sczlog**
1. create workflow yaml
 - the action will be fired every time a pull request or pull target main branch;
 - the action can be fired manually.
2. change registry to public one.
 - to make yarn can install package from correct registry

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,6 +3,8 @@
   "private": true,
   "scripts": {
     "lerna": "lerna",
+    "prepublish:ci": "lerna run prepublish",
```
Comment (Yuyz0112) [inline package.json:6]: Can be `prepublish`?

File: config/jest.ci.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+      branches: 0,
+      functions: 0,
+      lines: 0,
+      //   statements: -5,
+    },
+  },
+  reporters:
+    process.env.CI_COMMIT_REF_NAME === 'master' ||
+    process.env.CI_BUILD_REF_NAME === 'master'
+      ? ['default', '<rootDir>/config/testrail-reporter.js']
```
Comment (Yuyz0112) [inline config/jest.ci.js:26]: not related to this project?

File: config/jest.ci.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+const path = require('path');
+
+module.exports = {
+  moduleFileExtensions: ['ts', 'js'],
+  testEnvironment: 'node',
+  maxWorkers: '50%',
+  rootDir: process.cwd(),
+  // The UI and server packages should follow its jest config
```
Comment (Yuyz0112) [inline config/jest.ci.js:8]: not related to this project

Comment (Yuyz0112) [reply config/jest.ci.js:8]: Please check other config, such as maxWorkers

**Review by Yuyz0112 — CHANGES_REQUESTED** [review]

File: config/jest.ci.js (hunk 3 above)
Comment (Sczlog) [reply config/jest.ci.js]: @Yuyz0112 
Sorry about the mistake, I had removed some of the irrelevant config but still left some here.
I have updated the config for ci and modify some babel config for unit test, `maxWorkers:'50%'` was kept due to it is the best practice for jest to run in environments with variable CPUs available.
PTAL

**Comment by Yuyz0112** [comment]
good job!

**Review by Yuyz0112 — APPROVED** [review]

**Pull request merged**